pkg/security/tests/fim_test.go

```diff
@@ -75,3 +76,82 @@ func TestFIMOpen(t *testing.T) {
 		t.Error("Event received (rule is in write only mode, and the open is read only)")
 	})
 }
+
+func TestFIMPermError(t *testing.T) {
+	SkipIfNotAvailable(t)
+
+	ruleDefs := []*rules.RuleDefinition{
+		{
+			ID:         "test_perm_open_rule",
+			Expression: `open.file.path == "{{.Root}}/test-file"`,
+		},
+		{
+			ID:         "test_perm_unlink_rule",
+			Expression: `unlink.file.path == "{{.Root}}/test-file"`,
+		},
+	}
+
+	test, err := newTestModule(t, nil, ruleDefs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer test.Close()
+
+	testFile, _, err := test.Path("test-file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(testFile)
+
+	f, err := os.Create(testFile)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = f.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = os.Chown(testFile, 2002, 2002)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	syscallTester, err := loadSyscallTester(t, test, "syscall_tester")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	test.Run(t, "open", func(t *testing.T, _ wrapperType, cmdFunc func(cmd string, args []string, envs []string) *exec.Cmd) {
+		args := []string{
+			"process-credentials", "setuid", "4001", "4001", ";",
+			"open", testFile,
+		}
+		envs := []string{}
+
+		test.WaitSignal(t, func() error {
+			cmd := cmdFunc(syscallTester, args, envs)
+			_, _ = cmd.CombinedOutput()
+			return nil
+		}, func(_ *model.Event, rule *rules.Rule) {
+			assertTriggeredRule(t, rule, "test_perm_open_rule")
+		})
```

Comment:
We should test the return value too to ensure we got the "good" EPERM (or add it in the rule expression directly).
Same for unlink test.